documentation/internal/Actions-on-Applications.md

```diff
@@ -0,0 +1,59 @@
+# Actions on Applications
+
+Actions define what a user can do related to an existing application. The actions take in account the current user own permissions to `apply` and the application template_permission to `review` or `assign` based on the current `level` and `stage` of the review.
+So basically we will define actions related to user roles of: **Applicant**, **Assigner**, **Reviewer** and **Consolidator** (basically actions are the same as Reviewer).
+
+**Note**: The same user can have more than one user-role. e.g. Assigner and reviewer.
+**Note 2**: One restriction is to not have an Applicant user with other user-role. This combination isn't prevented on the server though.
+
+Let's have a look on the action definitions for any user-roles:
+
+![Actions on Applications Flow](images/Actions-on-Applications-Flow.png)
+
+## Applicant's actions
+
+Actions are available to the applicant user for their own applications, and very basic for now:
+
+- `Continue` -> Application created that hasn't been submitted
+- `View` -> Application that has been submitted or after application process is completed
+- `Update` -> After an application has been submitted and reveiwed, when there are **changes required** to the Applicant
+
+### Other applicant's actions (todo)
+
+- `Widthdrawn` -> Application that has been submitted
+- `Delete` -> Application that hasn't been submitted
+
+## Reveiwer's or Consolidator's actions
+
+Actions are available to the Reviewer user (based `template_permission`) after the **application is submitted**.
+Actions more specific to the Consolidator user are available after there is a review **stage increment** or after **review is submitted**.
+
+- `Continue` -> Review created that hasn't been submitted
+- `Start` -> Review has been assigned (or self-assigned)
+- `Self-Assign` -> Review hasn't been assigned to anyone (and permission to SELF_ASSIGNMENT applies)
+- `Re-Review` -> Review has been submitted and lower level Reviewer has submitted **changes requested** by you
+- `Update` -> Review has been submitted and higher level Reviewer has submitted **changes requested** to you
+- `View` -> Review has been submitted or after application process is completed
+
+## Assigner's actions
+
+The options are per section of the application, but will show on the application list as the top most possible one (combining the actions available for all sections in the application):
+
+- `Assign` -> When no reviewer is assigned yet (checks `review_question_assignment`)
+- `Re-assign` -> When reviewer is assigned
+
+**Note**: When re-assigning section(s) when the review has been started there is some pending work todo for moving assignments
+
+### Application list view
+
+On server side we use smart view to calculate the count of each of the following (used to define available actions on the front-end) based on status of `review_assignment` and `review` (associated to self for Reviewer, and associated to any review for Assigners).
```

Comment:
Could you perhaps specify what you mean by "associated to"? You mean using the "user" as the parameter for the back-end Function, right? Maybe just explain that a little more?

Reply:
I can rephrase this better : ) will do tomorrow morning though

Reply:
@CarlosNZ I tried to rephrase this, and actually realise it wasn't accurate, so hopefully now it's better to understand and makes sense: 2a1079f